src/pages/home/report/ReportActionItemSingle.js

```diff
@@ -111,7 +111,7 @@ function ReportActionItemSingle(props) {
 
     // If this is a report preview, display names and avatars of both people involved
     let secondaryAvatar = {};
-    const displayAllActors = props.action.actionName === CONST.REPORT.ACTIONS.TYPE.REPORTPREVIEW && props.iouReport;
+    const displayAllActors = useMemo(() => props.action.actionName === CONST.REPORT.ACTIONS.TYPE.REPORTPREVIEW && props.iouReport, [props.action.actionName, props.iouReport]);
```

Comment:
We should use it because - it's used inside deps for useCallback - const showActorDetails = useCallback(() => {,
and any new render will re-creates useCallback - which should not be

Reply:
`displayAllActors` would change depends on `props.action.actionName` and `props.iouReport` regardless the use of `useMemo`. If those props don't change, `displayAllActors` won't be changed.

Reply:
yes - and if displayAllActors (if we use useMemo) will not changed - useCallback will also not be changed
Without useMemo - displayAllActors will be as usual variable - which will be re-created every time - and will cause re-creation of useCallback

Reply:
useCallback is already re-creates every time on any new re-render because of some not memoized deps
https://github.com/Expensify/App/pull/24624/files#r1296846310

Reply:
@narefyev91 
Whether `displayAllActors` is re-created or not isn't important. What is important is its value. `useCallback` is created only when its dependencies change. Unless its value isn't changed, `useCallback` isn't re-created.

If this doesn't make sense, let's open a discussion on slack

Reply:
@s-alves10 hmmm can you please explain why it's not important?
React re-creates any variable on re-render if those variables are not wrapper inside useMemo and useCallback
if displayAllActors will not be wrapped in useMemo - it will be re-creates, and it used inside useCallback - which will cause re-creation of useCallback as well.
Feel free to raise a conversation - if you think it's necessary
![photo_2023-08-23 18 46 49](https://github.com/Expensify/App/assets/28352309/8ad59de7-1214-4e2a-8951-a63f49537e6c)
Using without memoization react will warn you about it (with standard message) if this rule exists in eslint:
`The 'foo' function makes the dependencies of useMemo Hook (at line 2) change on every render. Move it inside the useMemo callback. Alternatively, wrap the definition of 'foo' in its own useCallback() Hook.`

And also you can read here https://github.com/facebook/create-react-app/issues/6880#issuecomment-485912528

Reply:
@fedirjh 

So do you think even when `a` is a string type(value type) and has the same value, `useCallback` is recreated?
useCallback compares the value of its dependency. The value here means its value for value type variables, and ref for reference type variables

Reply:
Our case:
```
const b = a; // b changes only when a changes
const b = useMemo(() => a, [a]); // so we don't need to memoize
```
You are thinking like this:
```
const b = {...a}; // b changes even when a doesn't change
const b = useMemo(() => ({...a}), [a]); // so we need to memoize
```

Reply:
After some checks - i agree that simple passed variable will not trigger recreation of useCallback. Here can be 2 options either leave as it's now in case props.iouReport is object and useMemo needed here or refactor as @bernhardoj suggested to Boolean(props.iouReport) and remove useMemo. I think it's not critical

Reply:
If nothing is going to break, I think we can leave it as is and maybe we can clean up in the future. It's just semantics, I also missed it, I thought it was necessary, but it is also true you guys are probably more knowledgable about React that me 😄

Reply:
We good to jump on it later